### ISSUE TITLE
How to set-up the mapping?

### ISSUE THREAD

**Issue by rkorebrits** [infojunkie/MantisBT-Slack#12]
Hi there,
Thanks for the plugin! So far working nice, we just can't get 1 thing to work: "Slack Channels"

*Option name is plugin_Slack_channels and is an array of 'Mantis project name' => 'Slack channel name'. Array options must be set using the Configuration Report screen. *

What is the actual value we put into the the *Configuration Report screen*?

Users: All
Project Name: All
Configuration Option: plugin_Slack_channels
Type: Complex
Value: ???

We are trying to route the **Mantis** group "feno - Online Shop" to **Slack** "feno-online-shop".

Also, can we also send notification to groups, as opposed to channels?

Thanks in advance!

**Comment by infojunkie** [comment]
Here's an example of what we use in our configuration:
```php
array (
  'Checkdesk' => '#checkdesk-dev',
  'Sysops' => '#sysops',
  'Yallah/C2C' => '#yallah',
  'Verne' => '#verne-dev',
  'Bridge' => '#bridge-dev',
)
```
You can send notification to `@users` in addition to `#channels`.

**Comment by rkorebrits** [comment]
Thanks for your quick response!

Can you please confirm the other fields also?

When I use *Type* "Complex" and enter:

    array (
      'feno - Online Shop' => '#feno-online-shop-chan',
    )

Then in the overview it shows as:

    array (
      0 => 'feno',
      1 => 'Online',
      2 => 'Shop',
      3 => 'feno',
      4 => 'online',
      5 => 'shop',
      6 => 'chan',
    )

If I use the same sting but with type = "string" then in the overview I get:

    ' array (
    \'feno - Online Shop\' => \'#feno-online-shop-chan\',
    )'

In both cases it doesn't work.

**Comment by infojunkie** [comment]
Other fields look sane as well, especially the Complex type. 
* We're using MantisBT 1.2.8, what's your version?
* Are there any other complex types in your Mantis configuration page? Do they look different?

**Comment by rkorebrits** [comment]
* We have just upgraded to the v3b.
* We don't have other complex types in the config screen, but there are a few visible in the database (I think these are system arrays, e.g. "view_issues_page_columns")

Just checked the DB for what the field looks like there and there it shows as: `["feno","Online","Shop","feno","online","shop","chan"]`

I've changed this to `["feno - Online Shop","#feno-online-shop-chan"]`, but the in the config screen I see this: 

    array (
      0 => 'feno - Online Shop',
      1 => '#feno-online-shop-chan',
    )

It's doesn't seem to like setting your own array keys..

What should this look like? Otherwise I can directly change it in the DB..?

**Comment by infojunkie** [comment]
On my local database, the config looks like this:
```php
a:1:{s:6:"Mantis";s:7:"@kratib";}
```

**Comment by infojunkie** [comment]
"Slack Bot Name" can be any name you choose (e.g. 'Mantis') and "Slack Bot Icon" is either the URL of an image or an emoji, as per https://api.slack.com/incoming-webhooks (section "Customized Name and Icon"). Otherwise, your settings look correct. By "not working", I guess you mean you don't receive the notification on Slack. Do you see any error in the PHP log? Also, make sure you do have the PHP cURL extension loaded.

**Comment by greenblack3000** [comment]
Yes I do not receive the notifications in Slack. I do not see any error in the PHP log. Also I do have cURL extension enabled. Any other suggestions? Thanks.

**Comment by bitwombat** [comment]
> "Slack Bot Name" can be any name you choose (e.g. 'Mantis') and "Slack Bot Icon" is either the URL
> of an image or an emoji

* An emoji like ```:some-emoji:``` or like ```some-emoji```?

* Can it be blank?

* URL of an image on slack.com, or anywhere?  (I wasn't able to find a list of image URLs - their example is the only one, and directory listing is not allowed there).

**Comment by bitwombat** [comment]
From the code, is see it needs to be ```:some-emoji:`.

My problem was that the URL had leading and trailing spaces from copying it from Slack's webhooks page.  I'll issue a pull-request with a change which will hopefully yell about a failed curl_exec.
